### ISSUE TITLE
Maintainers for Q3, 2015

### ISSUE THREAD

**Issue by xordoquy** [tomchristie/django-rest-framework#2995]
This issue is for determining the maintenance team for the Q3, 2015 period.

Please see the [Project management](http://www.django-rest-framework.org/topics/project-management/) section of our documentation for more details.

---

#### Renewing existing members.

The following people are the current maintenance team. Please checkmark your name if you wish to continue to have write permission on the repository for the Q3, 2015 period.

- [x] @tomchristie
- [x] @xordoquy 
- [x] @carltongibson 
- [x] @kevin-brown 
- [x] @jpadilla

---

#### New members.

If you wish to be considered for this or a future date, please comment against this or subsequent issues.

To modify this process for future maintenance cycles make a pull request to the [project management](http://www.django-rest-framework.org/topics/project-management/) documentation.

**Comment by tomchristie** [comment]
Thanks @xordoquy I don't think I'd have remembered to do this! :)

**Comment by carltongibson** [comment]
PSA: Again, I'm happy to continue but if someone's reading this thinking "I'd like to join up" I'm very happy to step aside, since I can't really say I've had as much time to help as I'd of liked.

**Comment by tomchristie** [comment]
I'm more than happy with what you continue to contribute to the project @carltongibson, but appreciate the note. :smile:

**Comment by cancan101** [comment]
I would be interested in being a maintainer.

**Comment by carltongibson** [comment]
Hey @cancan101. I think you have the makings of a great maintainer for DRF: you've been around for a good while, you're active and you use it daily (or so it appears).

I have two preliminary suggestions: 

**1**: Get on StackOverflow, the Mailing List and the Issue Tracker here and help resolve the many questions that come up. Most issues are usage questions and an experienced user such as yourself can show people the solutions which perhaps aren't obvious. 90% of the time a targeted link to the docs is most of what's needed here. Most of the maintenance is triaging tickets. 

**2**: You need to make sure you have internalised, and are on-board with, the general approach to maintenance for DRF. 

I'll summarise as "Keep the base implementation simple and make it easy for people to override in user land if they need to" — others can elaborate there. 

There are any number of small  _improvements_ that could be made — "Add this feature here", "Add that feature there" — but the reality is we can't make them. Each addition adds to the maintenance burden and clouds the code base for people needing different behaviour. 

A feature freeze is on the cards: at least for now, DRF is nearly done. As such almost all "Add X" requests need to be answered with a "No". A collection of snippets for how to implement X yourself comes in handy. 

(This is not to say features can't be added. If they keep coming up then they can be considered, but "almost all...")

The need to say "No" here is what makes maintenance hard: in good will, users, including yourself, have ideas for API that would make their particular use-case easier. They then open a ticket requesting said API. We then have to think through why that API isn't appropriate and usually offer a solution. This is hard — in time if nothing else. (The reality is most times _issues_ are usage questions; most _features_ are "Override..." or "Implement ... yourself".) 

So, if you'd like to help maintain DRF, jump in! :-) Triage tickets; help solve user problems: that's the 90%. The _maintainer_ status follows from that: the example to follow is the machine-like @jpadilla, who appeared from nowhere, resolved ≈8000 tickets and was promptly awarded the commit-bit, which in fairness isn't all that important _per se_.

If you can do this then you'd be a great maintainer. As I've said I've not had masses of time so stepping aside isn't a problem for me. 

My concern right now is that I'm not sure you're yet on board with **2**. In good will I'm quite sure, you open a lot a tickets which fall into the category described above. 

Now — don't get me wrong — there's nothing wrong with opening tickets — if there are issues there are issues, and seeing what users hit up against, and what comes up time and again, is useful for judging the state of the project — **but** I'm not convinced you are (as yet) on board with the general approach to maintenance for DRF. 

Bottom line is prove me wrong: resolve tickets, answer questions, ease the maintenance burden. Demonstrate that I'm mistaken. If you really want the "Maintainer" status, target Q4. 

Here I'm not trying to upset or offend — it's all said with [_philia_](https://en.wikipedia.org/wiki/Philia) — I really do think you'd make a good maintainer.

**Comment by xordoquy** [comment]
The key point here to me is to make sure the one that merges the PR is different from the one that did it.

**Comment by carltongibson** [comment]
Yeah perhaps you're right. My concern is that most of it isn't really about PRs. 

@cancan101 is active and that's going to add more than my being AWOL most of the time. I'm thinking I should step down and he should step up. (Will untick appropriately.)

**Comment by cancan101** [comment]
I am certainly more than willing to participate in the review of PRs in addition to submitting fixes for open issues.

Separate even from my candidacy as a maintainer, I think @carltongibson raises a good point about documenting the DRF philosophy / stance on what features belong in core vs not. Ideally there would be some documented policies around requests for new features vs requests for API changes allowing others to more easily extend the functionality of DRF.

I appreciate that some of my early feature requests fell under the former category of feature creep on the core functionality of DRF. I believe that many of my newer feature requests are more in the latter category: requests for small changes to APIs that can allow other packages (for example drf-extensions) to more readily add functionality. There is certainly a cost of establishing a public API, so I do understand that even API changes that allow for extensions are not always desirable.

I do think that along the lines of some of @tomchristie's blog posts (for example his post on tickets: http://www.dabapps.com/blog/sustainable-open-source-management/), there is some opportunity here to better document policy the project's policy. This will allow maintainers to more quickly provide better feedback to feature requests by simply pointing the requestor to the policies.

**Comment by tomchristie** [comment]
Very well balanced view from @carltongibson - thank you.

> If you really want the "Maintainer" status, target Q4.

If @cancan101 and @carltongibson didn't have any major object then that'd probably be my preference. The key for improving our maintenance is that we each focus on closing more tickets than we open. I'd like to see anyone new to the team demonstrate that first.

> I think @carltongibson raises a good point about documenting the DRF philosophy / stance on what features belong in core vs not. Ideally there would be some documented policies around requests for new features vs requests for API changes allowing others to more easily extend the functionality of DRF.

Agreed.

> there is some opportunity here to better document policy the project's policy.

And again, agreed.

There's also lots of scope for better explaining how folks can help the maintainance of the project, and what we'd like to see demonstrated for membership of the maintenance team. We have a good process going at the moment, and I think a pretty good shared understanding on the team of it, but we've not documented it fully yet.

**Comment by tomchristie** [comment]
Leaving things as is for this quarter - see no need to expire anyone's write permissions right now.
